core/test/unit/adapters/storage/utils_spec.js

```diff
@@ -90,4 +90,76 @@ describe('storage utils', function () {
             result.should.be.equal('http://example-blog.com/ghost-logo.png');
         });
     });
+
+    describe('fn: isLocalImage', function () {
+        it('should return true when absolute URL and local file', function () {
+            var url = 'http://myblog.com/content/images/2017/07/ghost-logo.png',
+                result;
+
+            urlForStub = sandbox.stub(utils.url, 'urlFor');
+            urlForStub.withArgs('home').returns('http://myblog.com/');
+            urlGetSubdirStub = sandbox.stub(utils.url, 'getSubdir');
+            urlGetSubdirStub.returns('');
+
+            result = storageUtils.isLocalImage(url);
+            should.exist(result);
+            result.should.be.equal(true);
+        });
+
+        it('should return true when absolute URL with subdirectory and local file', function () {
+            var url = 'http://myblog.com/blog/content/images/2017/07/ghost-logo.png',
+                result;
+
+            urlForStub = sandbox.stub(utils.url, 'urlFor');
+            urlForStub.withArgs('home').returns('http://myblog.com/');
+            urlGetSubdirStub = sandbox.stub(utils.url, 'getSubdir');
+            urlGetSubdirStub.returns('/blog');
+
+            result = storageUtils.isLocalImage(url);
+            should.exist(result);
+            result.should.be.equal(true);
+        });
+
+        it('should return true when relative URL and local file', function () {
+            var url = '/content/images/2017/07/ghost-logo.png',
+                result;
+
+            urlForStub = sandbox.stub(utils.url, 'urlFor');
+            urlForStub.withArgs('home').returns('http://myblog.com/');
+            urlGetSubdirStub = sandbox.stub(utils.url, 'getSubdir');
+            urlGetSubdirStub.returns('');
+
+            result = storageUtils.isLocalImage(url);
+            should.exist(result);
+            result.should.be.equal(true);
+        });
+
+        it('should return true when relative URL and local file', function () {
+            var url = '/blog/content/images/2017/07/ghost-logo.png',
+                result;
+
+            urlForStub = sandbox.stub(utils.url, 'urlFor');
+            urlForStub.withArgs('home').returns('http://myblog.com/');
+            urlGetSubdirStub = sandbox.stub(utils.url, 'getSubdir');
+            urlGetSubdirStub.returns('/blog');
+
+            result = storageUtils.isLocalImage(url);
+            should.exist(result);
+            result.should.be.equal(true);
+        });
+
+        it('should return false when no local file', function () {
```

Comment:
Tests look good. The tests also pass with the old implementation of `isLocalImage`.
One other test case i can think of is ensuring that a local image get's detected independent of the protocol.
But other than that, looks good to merge for the next release on Tuesday 👍

Reply:
@AileenCGN Want to add that test case?

Reply:
@kirrg001 I did some work on that already, but IMO it would only make sense to check when the blog is set up as https and the requested image protocol is http. And for this case, I need to actually change the code a bit, as this test would not pass. 

What do you think?

Reply:
Exactly. I think it's a nice-to-have test, because Ghost redirects the request before, [see](https://github.com/TryGhost/Ghost/blob/master/core/server/middleware/url-redirects.js#L76). In worst case this case (or any case we can't think of atm) would not read from the local storage. It doesn't mean it breaks or fails. Maybe it's fine to add this test case with a `.skip` and leave a comment.

Reply:
@kirrg001 Added a test (`.skip` currently) with a comment and a todo